hudi-client/src/main/java/org/apache/hudi/async/AbstractAsyncService.java

```diff
@@ -39,9 +39,9 @@
   private static final Logger LOG = LogManager.getLogger(AbstractAsyncService.class);
 
   // Flag to track if the service is started.
-  private boolean started;
+  private volatile boolean started;
```

Comment:
Sorry, I don't get the point here, would you please describe it in more details why need this change?

Reply:
The purpose of this change is ensure the visibility of `started` and `shutdownRequested`.

The getter method `isStarted()` provide a way to read the service status.

Reply:
@dugenkui03 hi, sorry for the late reply. Is it necessary to change to volatile and what's the problem of current approach?

Reply:
I think `started ` will be read by multiple thread, so it is necessary to guarantee the visibility of `start`. As well as `shutdownRequested `.

May be I misunderstand the usage scenario of them.

I convert this PR to draft for the moment.

Reply:
> I think `started ` will be read by multiple thread, so it is necessary to guarantee the visibility of `start`. As well as `shutdownRequested `.
> 
> May be I misunderstand the usage scenario of them. I convert this PR to draft for the moment, until i make sure that this change is necessary or not.

OK

Reply:
I think it makes sense to do this. `volatile` will ensure the other thread sees the update right away